File: solutions/900.reordered-power-of-2/reordered-power-of-2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int N) {
        set<string> two;
        for(int i = 0; i < 32; i++) {
            string s = to_string(1LL << i);
            sort(s.begin(), s.end());
            two.insert(s);
        }
        string s = to_string(N);
        sort(s.begin(), s.end());
        return two.find(s) != two.end();
    }
};

File: solutions/900.reordered-power-of-2/reordered-power-of-2.java
class Solution {
    public boolean reorderedPowerOf2(int N) {
        long c = counter(N);
        for(int i = 0; i < 32; i++) 
            if(counter(1 << i) == c) 
                return true;
        return false;
    }
    private long counter(int n) {
        long res = 0l;
        while(n > 0) {
            res += (int)Math.pow(10, n % 10);
            n /= 10;
        }
        return res;
    }
}